class-21.md

```diff
@@ -0,0 +1,44 @@
+# Intro to Blockchain Development
+
+### Class 21 - Alternative Consensus Mechanisms
+
+It is very difficult to attack a blockchain at the protocol level. Generally, most attacks happen at the the level of a user, wallet or exchange. We'll run over some: 
+
+Reading & Videos:
+
+- [Ripple Consensus Whitepaper](https://ripple.com/files/ripple_consensus_whitepaper.pdf) - Whitepaper about ripple's consensus mechanism
+- [IOTA Whitepaper](https://assets.ctfassets.net/r1dr6vzfxhev/2t4uxvsIqk0EUau6g2sw0g/45eae33637ca92f85dd9f4a3a218e1ec/iota1_4_3.pdf) - IOTA's whitepaper outlining how a DAG works
+- [Basic Explaination of a DAG](https://stackoverflow.com/questions/2283757/can-someone-explain-in-simple-terms-to-me-what-a-directed-acyclic-graph-is) - Helpful StackOverflow example explaining the subject
+
+### Going Beyond Blocks
+
+Instead of finding different ways to optimize how data is persisted in blocks, work functions, block sizes and block production rates, we can also look for room to optimize in how data is stored and validated. Instead of batching transactions into blocks and tracking the previous block in a linked list, we can add transactions asynchronously. Instead of validating all transactions previously in each block, a block can just validate a few that came before it. Eventually, as a transaction has been represented enough by the network, it can reach consensus.
```

Comment:
`...blocks, work functions, block sizes and block production rates...`
-> Forgot the oxford comma here.

`Eventually, as a transaction has been represented enough by the network, it can reach consensus.`
-> Unnecessary comma here.
-> `Eventually when a transaction has been represented enough number of times by the network, it can reach consensus.`